you-have-mail-android/app/src/main/java/dev/lbeernaert/youhavemail/screens/Settings.kt

```diff
@@ -54,57 +60,60 @@ fun Settings(
                 .padding(20.dp)
                 .fillMaxSize()
         ) {
+            val timeIntervals =
+                listOf(30UL, 60UL, 120UL, 180UL, 300UL, 600UL, 900UL, 1200UL, 1800UL, 3600UL)
             var expanded by remember { mutableStateOf(false) }
-            val items = listOf(60UL, 150UL, 300UL, 600UL, 900UL, 1800UL, 3600UL)
-            var selectedIndex by remember { mutableStateOf(0) }
+            var selectedIndex by remember { mutableIntStateOf(0) }
 
             val onPollIntervalModified: () -> Unit = {
                 runTask(updatingPollIntervalLabel) {
-                    onPollIntervalUpdate(items[selectedIndex])
+                    onPollIntervalUpdate(timeIntervals[selectedIndex])
                 }
             }
 
             Text(
                 text = stringResource(id = R.string.poll_interval),
-                style = MaterialTheme.typography.caption
+                textAlign = TextAlign.Center,
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.fillMaxWidth()
             )
-            Spacer(modifier = Modifier.padding(10.dp))
+            Spacer(modifier = Modifier.padding(5.dp))
             Text(
                 text = stringResource(id = R.string.poll_interval_desc),
                 style = MaterialTheme.typography.subtitle1
             )
-            Spacer(modifier = Modifier.padding(10.dp))
+            Spacer(modifier = Modifier.padding(5.dp))
             TextField(
                 modifier = Modifier
-                    .fillMaxWidth(),
-                value = secondsToText(
-                    pollIntervalValue,
-                    secondsStr,
-                    minutesStr
-                ),
+                    .fillMaxWidth()
```

Comment:
I have nothing against the new way the drop down works, but the colors aren't exactly jiving together.

![Screenshot_20241210-104833](https://github.com/user-attachments/assets/3bbca814-fe94-4578-aa51-8b3d38c8093c)

Then there is also an issue with the drop down menu where it does not expand as far as it would previously:

![Screenshot_20241210-104827](https://github.com/user-attachments/assets/e74081df-49b5-48ed-9712-43bdc7654dae)

Could you also revert back to the previous background color? I think it is easier to read the white text.

Reply:
oops, sorry. I tested this application only on smartphones with bright mode set in Android settings. It seems that system-wide dark mode became available since Android 10 (Sdk 29)
So now I made a change that there is forced gray background of dropdownmenu to gray (it was probably overridden by system-wide dark mode theme to black) and text inside dropdownmenu is also now forced to black (it was probably overridden by system-wide dark mode theme to white)
colors = TextFieldDefaults.textFieldColors(backgroundColor = colorResource(id = R.color.gray_dropdownmenu)),
Dropdownmenu should look better now on both dark mode and bright mode. You can check it if You want. If this still look bad then I can change or revert the colors. Maybe there is some kind of switch for themes in application that get activated depending on whether dark or bright mode is selected in system-wide Android settings

hmm, width of dropdownmenu can be expanded as previously with this DropdownMenu(modifier = Modifier.fillMaxWidth())
I just thought that this currently contracted (.wrapContentWidth()) dropdownmenu might look more cute and not take the whole size of screen, and it can be easier to quit/escape this expanded dropdownmenu after changing the mind. I can revert that if You want. 

Below I attached some screenshots of how this looks like on my smartphone with system-wide bright mode

Reply:
![Screenshot_20241115-135808](https://github.com/user-attachments/assets/77d8d1df-98b4-41d4-a1ea-29f0f19aad2a)
![Screenshot_20241211-171331](https://github.com/user-attachments/assets/7172bb87-716f-4879-87de-f3b49281bf6a)
![Screenshot_20241211-171340](https://github.com/user-attachments/assets/ced92393-9347-449a-a53d-8b40a658f66e)

Reply:
I would rather see the drop down menu take up the whole width of the item it drops down from instead.

In dark mode the input is now white background with black text. Can you restore the original color setup in this mode?

Reply:
ok, of course, I reverted all colors to original ones from your last commit.
ok, I added a tweak to DropDownMenu width so that it has identical width as TextField it drops down from. So it is neither so thin and neither takes whole screen width
DropdownMenu(modifier = Modifier.width(with(LocalDensity.current) { textFieldSize.width.toDp() }))

Below I attached a screenshots how this Settings screen looks now on my Smartphone with system-wide bright mode set

Reply:
![Screenshot_20241214-145030](https://github.com/user-attachments/assets/e8f4bcaf-bcdd-4142-8f2b-43caf0b42e48)
![Screenshot_20241214-145035](https://github.com/user-attachments/assets/2539b758-6b32-4de9-8da2-348e8593485c)

Reply:
We are almost there :).

The one thing that is still off in dark mode is the text for the selected option. It is now in a shade of light grey when it was supposed to be white. (See attached screenshot).

The reset looks perfect :heart:

![Screenshot_20241215-112524](https://github.com/user-attachments/assets/f3ce7db3-0685-48ed-a2cf-e256bf7ac1d7)

Reply:
oops, sorry, I forgot that changing TextField to disabled also changed its text color to gray (surprisingly setting it to disabled before somehow allowed it to be clickable)
ok, I refactored this DropDownMenu a bit now so that all colors should be now reverted and identical as in your last commit. After refactor this TextField somehow can now be enabled and clickable simultaneously. trailingIcon is also clickable and DropDownMenu width matches TextField width

I attached some screenshots how this looks now :)

If there are other issues then you can tell me

Reply:
![Screenshot_20241215-200644](https://github.com/user-attachments/assets/6b1f3012-be60-41f1-9825-f191b88e543e)
![Screenshot_20241215-200648](https://github.com/user-attachments/assets/bc91ce01-4fc0-478a-baef-374ac800b0cb)
![Screenshot_20241215-200658](https://github.com/user-attachments/assets/6f199945-b01b-4680-a101-22fcc20b65a0)

Reply:
Thanks :)